added switch to show only issues in rdiff

lethliel opened the pull request:
new switch (--issues-only) which will only show the issues that differ.

marcus-h commented:
Just a very quick review:

> diff --git a/osc/commandline.py b/osc/commandline.py
> index 56bcfbc..f470422 100644
> --- a/osc/commandline.py
> +++ b/osc/commandline.py
> @@ -3753,6 +3753,8 @@ def do_diff(self, subcmd, opts, *args):
>      @cmdln.option('--oldpkg', metavar='OLDPKG',
>                    help='package to compare against'
>                    ' (deprecated, use 3 argument form)')
> +    @cmdln.option('--issues-only', action='store_true',
> +                        help='show only issues in diff')
>      @cmdln.option('-M', '--meta', action='store_true',
>                          help='diff meta data')
>      @cmdln.option('-r', '--revision', metavar='N[:M]',
> @@ -3842,9 +3844,12 @@ def do_rdiff(self, subcmd, opts, *args):
>                              old_project, old_package, rev1,
>                              new_project, new_package, rev2, not opts.plain, opts.missingok,
>                              meta=opts.meta,
> -                            expand=not opts.unexpand)
> -
> -        run_pager(rdiff)
> +                            expand=not opts.unexpand,
> +                            onlyissues=opts.issues_only)
> +        if opts.issues_only:
> +            print(rdiff)

Or run_pager?

> +        else:
> +            run_pager(rdiff)
>  
>      def _pdiff_raise_non_existing_package(self, project, package, msg = None):
>          raise oscerr.PackageMissing(project, package, msg or '%s/%s does not exist.' % (project, package))
> diff --git a/osc/core.py b/osc/core.py
> index fca78f4..3f5a0bc 100644
> --- a/osc/core.py
> +++ b/osc/core.py
> @@ -4649,7 +4649,7 @@ def get_source_file_diff(dir, filename, rev, oldfilename = None, olddir = None,
>  def server_diff(apiurl,
>                  old_project, old_package, old_revision,
>                  new_project, new_package, new_revision,
> -                unified=False, missingok=False, meta=False, expand=True, full=True):
> +                unified=False, missingok=False, meta=False, expand=True, onlyissues=False, full=True):
>      query = {'cmd': 'diff'}
>      if expand:
>          query['expand'] = 1
> @@ -4670,21 +4670,31 @@ def server_diff(apiurl,
>      if full:
>          query['filelimit'] = 0
>          query['tarlimit'] = 0
> +    if onlyissues:
> +        query['onlyissues'] = 1
> +        query['view'] = 'xml'
> +        query['unified'] = 0
>  
>      u = makeurl(apiurl, ['source', new_project, new_package], query=query)
> -
>      f = http_POST(u)
> +    if onlyissues:
> +      issue_list = []
> +      root = ET.fromstring(f.read())
> +      for node in root.findall('issues'):

Hmm I think there is exactly one "issues" node, isn't it?

> +          for issuenode in node.findall('issue'):
> +              issue_list.append('%s' % issuenode.get('label'))

Useless string formatting: just issue_list.append(issuenode.get('label'))

> +      return('\n'.join(issue_list))

Useless parentheses.

"Wrong" indentation (if block).

lethliel commented:
On 05/10/2017 10:12 PM, Marcus Hüwe  wrote:
> Just a very quick review:
> 

Thanks :) 

>> diff --git a/osc/commandline.py b/osc/commandline.py
>> - run_pager(rdiff)
>> + expand=not opts.unexpand,
>> + onlyissues=opts.issues_only)
>> + if opts.issues_only:
>> + print(rdiff)
> 
> Or run_pager?
> 

I switched to a normal print on purpose. The return is just a list of issue labels. So 
I don't see the need of a pager here. 

>> + else:
>> + run_pager(rdiff)
>>

<SNIP>

>> + root = ET.fromstring(f.read())
>> + for node in root.findall('issues'):
> 
> Hmm I think there is exactly one "issues" node, isn't it?
> 

Yes. I now use: 

node = root.find('issues')

>> + for issuenode in node.findall('issue'):
>> + issue_list.append('%s' % issuenode.get('label'))
> 
> Useless string formatting: just issue_list.append(issuenode.get('label'))
> 
>> + return('\n'.join(issue_list))
> 
> Useless parentheses.
> 
> "Wrong" indentation (if block).
>

I fixed the indentation and removed the useless string formatting and parentheses.
